### PR TITLE
Add decorator sdc_register_jitable

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
This PR adds decorator `sdc_register_jitable` which wraps `numba.extending.register_jitable` decorator.
`sdc_register_jitable` takes default parameters for `parallel` and `inline` from `SDC_AUTO_PARALLEL` and `SDC_AUTO_INLINE` environment variables.

I am planning to use it in #541 and #535 and other PRs for improving scalability.

### CODE REVIEW

**Comment by PokhodenkoSA** [comment]
Replaced all `register_jitable` in SDC with `sdc_register_jitable`.
Most `@register_jitable` was used in @densmirn series rolling functions. @densmirn is it ok to replace them?

**Comment by densmirn** [comment]
> Replaced all `register_jitable` in SDC with `sdc_register_jitable`.
> Most `@register_jitable` was used in @densmirn series rolling functions. @densmirn is it ok to replace them?

I think so.

File: sdc/utils.py [hunk 1]
```diff
@@ -591,6 +592,20 @@ def sdc_overload(func, jit_options={}, parallel=None, strict=True, inline=None):
     return overload(func, jit_options=jit_options, strict=strict, inline=inline)
 
 
+def sdc_register_jitable(*args, **kwargs):
+    updated_kwargs = kwargs.copy()
+    updated_kwargs['parallel'] = updated_kwargs.get('parallel', config_use_parallel_overloads)
+    updated_kwargs['inline'] = updated_kwargs.get('inline', 'always' if config_inline_overloads else 'never')
```
Comment (AlexanderKalistratov) [inline sdc/utils.py]: So, does it work?

Comment (AlexanderKalistratov) [reply sdc/utils.py]: I'm 90% sure it doesn't

Comment (PokhodenkoSA) [reply sdc/utils.py]: Yes. If I use e.g. `Always` it prints this:
image

Comment (PokhodenkoSA) [reply sdc/utils.py]: @AlexanderKalistratov could you propose how to check precisely?

Comment (AlexanderKalistratov) [reply sdc/utils.py]: The best way is to check IR with enabled/disable option

Comment (AlexanderKalistratov) [reply sdc/utils.py]: It seems, it doesn't work
https://github.com/numba/numba/issues/5142

Comment (PokhodenkoSA) [reply sdc/utils.py]: I have applied proposal patch from the issue.

**Pull request merged**